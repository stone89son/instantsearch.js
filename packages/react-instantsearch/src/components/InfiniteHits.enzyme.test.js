/* eslint-env jest, jasmine */

import React from 'react';

import InfiniteHits from './InfiniteHits';

import {mount, render} from 'enzyme';

describe('InfiniteHits', () => {
  it('accepts a itemComponent prop', () => {
    const hits = [{objectID: 0}, {objectID: 1}, {objectID: 2}];
    const Hit = 'Hit';
    const tree = render(
      <InfiniteHits
        itemComponent={Hit}
        hits={hits}
        isLastPage={false}
        refine={() => undefined}
      />
    );
    expect(tree).toMatchSnapshot();
  });

  it('calls refine when the load more button is clicked', () => {
    const mockedRefine = jest.fn();
    const hits = [{objectID: 0}, {objectID: 1}, {objectID: 2}];
    const Hit = 'Hit';
    const wrapped = mount(
      <InfiniteHits
        refine={mockedRefine}
        itemComponent={Hit}
        hits={hits}
        isLastPage={false}
      />
    );
    expect(mockedRefine.mock.calls.length).toBe(0);
    wrapped.find('button').simulate('click');
    expect(mockedRefine.mock.calls.length).toBe(1);
  });

  it('Button is disabled when it is the last page', () => {
    const hits = [{objectID: 0}, {objectID: 1}, {objectID: 2}];
    const Hit = 'Hit';
    const wrapped = mount(
      <InfiniteHits
        refine={() => undefined}
        itemComponent={Hit}
        hits={hits}
        isLastPage={true}
      />
    );
    expect(wrapped.find('button').props().disabled).toBe(true);
  });
});
