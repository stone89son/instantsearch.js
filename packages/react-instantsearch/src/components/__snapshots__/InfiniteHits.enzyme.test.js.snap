exports[`InfiniteHits accepts a itemComponent prop 1`] = `
 {
  "0": Object {
    "attribs": Object {},
    "children": Array [
      Object {
        "attribs": Object {
          "class": "root",
        },
        "children": Array [
          Object {
            "attribs": Object {},
            "children": Array [],
            "name": "hit",
            "next": Object {
              "attribs": Object {},
              "children": Array [],
              "name": "hit",
              "next": Object {
                "attribs": Object {},
                "children": Array [],
                "name": "hit",
                "next": Object {
                  "attribs": Object {},
                  "children": Array [
                    Object {
                      "data": "Load more",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "button",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                },
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
              },
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
          },
          Object {
            "attribs": Object {},
            "children": Array [],
            "name": "hit",
            "next": Object {
              "attribs": Object {},
              "children": Array [],
              "name": "hit",
              "next": Object {
                "attribs": Object {},
                "children": Array [
                  Object {
                    "data": "Load more",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "button",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
              },
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
            },
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {},
              "children": Array [],
              "name": "hit",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
            },
            "type": "tag",
          },
          Object {
            "attribs": Object {},
            "children": Array [],
            "name": "hit",
            "next": Object {
              "attribs": Object {},
              "children": Array [
                Object {
                  "data": "Load more",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "button",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
            },
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {},
              "children": Array [],
              "name": "hit",
              "next": [Circular],
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {},
                "children": Array [],
                "name": "hit",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
              },
              "type": "tag",
            },
            "type": "tag",
          },
          Object {
            "attribs": Object {},
            "children": Array [
              Object {
                "data": "Load more",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "button",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {},
              "children": Array [],
              "name": "hit",
              "next": [Circular],
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {},
                "children": Array [],
                "name": "hit",
                "next": [Circular],
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {},
                  "children": Array [],
                  "name": "hit",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                },
                "type": "tag",
              },
              "type": "tag",
            },
            "type": "tag",
          },
        ],
        "name": "div",
        "next": null,
        "parent": null,
        "prev": null,
        "root": [Circular],
        "type": "tag",
      },
    ],
    "name": "root",
    "next": null,
    "parent": null,
    "prev": null,
    "type": "root",
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xmlMode": false,
  },
}
`;
